spec/mailers/supervisor_mailer_spec.rb

```diff
@@ -134,6 +134,14 @@
           expect(mail.body.encoded).not_to match("volunteers have not signed in or created case contacts in the last 30 days")
         end
       end
+
+      context "volunteer from different org do not show" do
+        it "display no_recent_sign_in section" do
+          expect(mail.body.encoded).to match("volunteers have not signed in or created case contacts in the last 30 days")
+          expect(mail.body.encoded).to match(volunteer.display_name)
+          expect(mail.body.encoded).not_to match(volunteer_for_other_org.display_name)
+        end
+      end
```

Comment:
Does this code differ from the code right above it on lines 121-126?

```rb
it "display no_recent_sign_in section" do
  expect(mail.body.encoded).to match("volunteers have not signed in or created case contacts in the last 30 days")
  expect(mail.body.encoded).to match(volunteer.display_name)
  expect(mail.body.encoded).not_to match(volunteer_for_other_supervisor_same_org.display_name)
  expect(mail.body.encoded).not_to match(volunteer_for_other_org.display_name)
end
```

It feels like maybe the issue should have actually been closed earlier and the tests already existed?

Reply:
I think so. At least from my understanding, tests are meant to check behaviour not implementation.
Maybe it needs trying to reproduce the behaviour that the original PR was meant to fix and thereafter writing tests that confirm the behaviour is now what we want it to be.

Another thing we can check is if in the original bug report, these tests were not failing already.

@FireLemons's suggestion on Discord was:

> The goal of the test is to check that the PR from earlier works so if your test fails without the changes from the PR and passes with the changes it should be good.
> It would be nice to have proof that the test fails without the PR in your PR.

so that's the direction I'm going to explore.

Reply:
> The goal of the test is to check that the PR from earlier works so if your test fails without the changes from the PR and passes with the changes it should be good.

I would agree with that assessment. So sounds like you have a direction to explore. :+1:

Reply:
Hey @elasticspoon 

So I've tried to experiment with this and whereas going by [the conversation in the issue](https://github.com/rubyforgood/casa/issues/5597), it **_looks_** to be fixed; on the other hand, the addition of the filter doesn't break the existing tests. 

```
# original
 @inactive_volunteers = supervisor.inactive_volunteers

# change
 @inactive_volunteers = supervisor.inactive_volunteers.where(casa_org_id: @casa_organization.id)
```

which is technically covered by the following expectations:

```
expect(mail.body.encoded).to match("volunteers have not signed in or created case contacts in the last 30 days")
expect(mail.body.encoded).to match(volunteer.display_name)
expect(mail.body.encoded).not_to match(volunteer_for_other_supervisor_same_org.display_name)
expect(mail.body.encoded).not_to match(volunteer_for_other_org.display_name)
```

since the filter makes sure that only inactive volunteers **_from_** the same org as the supervisor are returned. The query as is already returns volunteers who are being supervised by the supervisor in question. 

I tried doing

```
 let!(:volunteer_2) do
    create(:volunteer, casa_org: other_org, supervisor: supervisor, last_sign_in_at: 31.days.ago)
end
```

and adding the expectation `expect(mail.body.encoded).not_to match(volunteer_for_other_org.display_name)` but that still passed with or without the added filter.

If you can reproduce the behaviour, that would be helpful. For me, I've hit the end of what I can try in order to reproduce the issue.

cc @FireLemons

Reply:
> I tried doing
>
>```
> let!(:volunteer_2) do
>    create(:volunteer, casa_org: other_org, supervisor: supervisor, last_sign_in_at: 31.days.ago)<
> end
>```
>
>and adding the expectation expect(mail.body.encoded).not_to match(volunteer_for_other_org.display_name) but that still passed with or without the added filter.

Did you end up using the `volunteer_2` that you created? It isn't present in the code you typed out and I wanna double check that.

Overall, though it sounds like we can close the issue.

Reply:
> Did you end up using the volunteer_2 that you created? It isn't present in the code you typed out and I wanna double check that.

Yes I did. Sorry for the confusion regarding the snippet 🙃

Reply:
Asked on Slack on how to reproduce the issue and got the following response from @compwron 

> Hi Leni!  

> This looks similar to my (one and only) fix on this project - https://github.com/rubyforgood/casa/pull/5571/files
From looking at the code, I think that bug was caused by the same issue, and fixed by my PR?  That would explain the issues reproducing it.

[Link to message on Slack](https://rubyforgood.slack.com/archives/CVB0QJGVD/p1726865472369729?thread_ts=1726862640.318119&cid=CVB0QJGVD)